server/src/main/java/org/opensearch/indices/replication/SegmentReplicationTargetService.java

```diff
@@ -38,7 +39,7 @@
  *
  * @opensearch.internal
  */
-public final class SegmentReplicationTargetService implements IndexEventListener {
```

Comment:
I think this is because you want to mock/spy on a this class in your tests correct?  We should look at maybe using mockito-inline instead?